### ISSUE TITLE
ColorMap getLookupTable fails when alpha=None

### ISSUE THREAD

**Issue by wailashi** [pyqtgraph/pyqtgraph#1009]
### Short description
The `getLookupTable` method of `pyqtgraph.ColorMap` fails to automatically determine whether alpha values should be included in a lookup table.

### Code to reproduce

```python
import pyqtgraph as pg

colors = [
    (0, 0, 0),
    (45, 5, 61),
    (84, 42, 55)
]
cmap = pg.ColorMap(pos=(0, 1, 2), color=colors)

cmap.getLookupTable(alpha=False)  # Works
cmap.getLookupTable(alpha=True)  # Works, but returned lookup table does not contain alpha values.
cmap.getLookupTable()  # Does not work
```

### Expected behavior
A lookup table is returned.

### Real behavior
The following exception is thrown:

```
Traceback (most recent call last):
  File "test.py", line 13, in <module>
    cmap.getLookupTable()  # Does not work
  File "...\venv\lib\site-packages\pyqtgraph\colormap.py", line 223, in getLookupTable
    alpha = self.usesAlpha()
  File "...\venv\lib\site-packages\pyqtgraph\colormap.py", line 236, in usesAlpha
    return np.any(self.color[:,3] != max)
IndexError: index 3 is out of bounds for axis 1 with size 3
```

### Tested environment(s)

 * PyQtGraph version: 0.11.0.dev0+gad7453c
 * Qt Python binding: 0.11.0.dev0+gad7453c
 * Python version: 3.7.3
 * NumPy version: 0.11.0.dev0+gad7453c
 * Operating system: Windows 10
 * Installation method: pip

**Comment by 2xB** [comment]
The issue here is that `pg.ColorMap` accepts RGBA values for `color`, but it gets RGB values.
This would not be a bug itself, if the example `ImageView.py` would not do the exact same thing:

> ImageView.py:
> ```python3
> colors = [
>     (0, 0, 0),
>     (45, 5, 61),
>     (84, 42, 55),
>     (150, 87, 60),
>     (208, 171, 141),
>     (255, 255, 255)
> ]
> cmap = pg.ColorMap(pos=np.linspace(0.0, 1.0, 6), color=colors)
> ```

There are now two obvious reactions:
a) Change the example to use RGBA
b) Automatically transform RGB to RGBA values in `ColorMap.__init__`

I suggest to iterate over `color` and parse the entries with `pg.functions.mkColor`, which would make the API more consistent. What do others think of this?

**Comment by 2xB** [comment]
@wailashi You are probably helped with the information that you should currently initialize `pg.ColorMap` with RGBA values?

**Comment by wailashi** [comment]
I came across this when trying to get a lookup table from a HistogramLUTWidget which I am using to change the colours on a scatter plot.

I was trying to do `pg.HistogramLUTWidget.gradient.colorMap().getLookupTable(mode='qcolor')`. This fails with the same error as in my first post.

While looking into this I found `pg.HistogramLUTWidget.gradient.getLookupTable()` which has a different method signature and does not return QColors.

[
Here is a gis](https://gist.github.com/wailashi/6ea42db5ac78f6af7a8f10a7771a13bc)t with a modified HistogramLUT.py from the examples directory with the failing getLookupTable calls.

For my use case `pg.HistogramLUTWidget.gradient.colorMap().getLookupTable(mode='qcolor', alpha=True)` is sufficient as I am not interested in the alpha values.
